### PR TITLE
Delete bucket

### PR DESCRIPTION
**Pull request by inaod568**
Delete the bucket and its contents when deleting / archiving the project. The bucket needs to be empty in order to delete it. 

- [x] Tests passing
- [x] Black formatting
- [ - ] Migrations for any changes to the database schema
- [ - ] Rebase/merge the `dev` branch
- [ - ] Note in the CHANGELOG

### CODE REVIEW

File: dds_web/api/api_s3_connector.py [hunk 1]
```diff
@@ -84,11 +85,21 @@ def get_s3_info(self):
...
+        # Get bucket object
         bucket = self.resource.Bucket(self.project.bucket)
+
+        # Delete objects first
         bucket.objects.all().delete()
 
+        # Delete bucket
+        bucket.delete()
+
+        # Clean up
```
Comment (talavis) [inline dds_web/api/api_s3_connector.py:99]: Is this really needed?

Comment (inaod568) [reply dds_web/api/api_s3_connector.py:99]: The clean up? It doesn't hurt, the bucket doesn't exist so there's no need to keep it in memory

Comment (talavis) [reply dds_web/api/api_s3_connector.py:99]: The effect should be the same if you just set it to None, or even just leave it.

Comment (inaod568) [reply dds_web/api/api_s3_connector.py:99]: Yes, basically the same. So I will keep this. Plus this is the method we use in multiple other places in the code.

**Review by talavis — APPROVED** [review]

**Pull request merged**